generator/src/test/java/com/scottlogic/deg/generator/cucumber/utils/InMemoryViolatingOutputTarget.java

```diff
@@ -0,0 +1,61 @@
+package com.scottlogic.deg.generator.cucumber.utils;
+
+import com.google.inject.Inject;
+import com.scottlogic.deg.generator.ProfileFields;
+import com.scottlogic.deg.generator.outputs.GeneratedObject;
+import com.scottlogic.deg.generator.outputs.targets.OutputTarget;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Defines an output target which stores the output data into the test state.
+ */
+public class InMemoryViolatingOutputTarget implements OutputTarget {
```

Comment:
Do we need a separate class for violations?

Reply:
with my current Guice knowledge, yes - to allow tests to pass in violating and normal modes.